# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1467308956 14400
#      Thu Jun 30 13:49:16 2016 -0400
# Node ID f1136cf44edce32dc6e282da4aa2167ac7c73550
# Parent  f078dd2c6f276e6ff81cfbf27974bf8a644632fc
fix linking of DLD functions with static build

* libinterp/module.mk (LIBINTERP_DLDFCN_LIBADD): New variable.
(libinterp_libinterp_la_LIBADD): Use it.

diff --git a/libinterp/module.mk b/libinterp/module.mk
--- a/libinterp/module.mk
+++ b/libinterp/module.mk
@@ -130,19 +130,21 @@ include libinterp/corefcn/module.mk
 include libinterp/dldfcn/module.mk
 
 $(srcdir)/libinterp/dldfcn/module.mk: $(srcdir)/libinterp/dldfcn/config-module.sh $(srcdir)/libinterp/dldfcn/config-module.awk $(srcdir)/libinterp/dldfcn/module-files
 	$(AM_V_GEN)$(SHELL) $(srcdir)/libinterp/dldfcn/config-module.sh $(srcdir)
 
 if AMCOND_ENABLE_DYNAMIC_LINKING
   OCT_FILES = $(DLDFCN_LIBS:.la=.oct)
   DLD_LIBOCTINTERP_LIBADD = libinterp/liboctinterp.la
+  LIBINTERP_DLDFCN_LIBADD =
 else
   OCT_FILES =
   DLD_LIBOCTINTERP_LIBADD =
+  LIBINTERP_DLDFCN_LIBADD = $(DLDFCN_LIBS)
 endif
 
 libinterp_liboctinterp_la_SOURCES = \
   libinterp/octave.cc \
   libinterp/version.cc \
   $(LIBINTERP_OPERATORS_SRC) \
   $(TEMPLATE_INST_SRC)
 
@@ -158,16 +160,17 @@ nodist_libinterp_liboctinterp_la_SOURCES
   libinterp/build-env-features.cc \
   libinterp/liboctinterp-build-info.cc \
   libinterp/version.h
 
 libinterp_liboctinterp_la_LIBADD = \
   libinterp/octave-value/liboctave-value.la \
   libinterp/parse-tree/libparse-tree.la \
   libinterp/corefcn/libcorefcn.la \
+  $(LIBINTERP_DLDFCN_LIBADD) \
   liboctave/liboctave.la \
   $(LIBOCTINTERP_LINK_DEPS)
 
 # Increment these as needed and according to the rules in the libtool manual:
 libinterp_liboctinterp_current = 3
 libinterp_liboctinterp_revision = 0
 libinterp_liboctinterp_age = 0
 
